shell/browser/native_window_views.cc

```diff
@@ -530,6 +530,23 @@ void NativeWindowViews::Unmaximize() {
 }
 
 bool NativeWindowViews::IsMaximized() {
+  // For window without WS_THICKFRAME style, we can not call IsMaximized().
+  // This path will be used for transparent windows as well.
+
+#if defined(OS_WIN)
+  gfx::Rect current_bounds = GetBounds();
+  if (!(::GetWindowLong(GetAcceleratedWidget(), GWL_STYLE) & WS_THICKFRAME)) {
+    // Compare the size of the window with the size of the display
+    auto display = display::Screen::GetScreen()->GetDisplayNearestWindow(
+        GetNativeWindow());
+    // Maximized if the window is the size of the display and in the top left
+    // corner
+    return ((display.work_area().width() == current_bounds.width()) &&
+            (display.work_area().height() == current_bounds.height()) &&
+            (GetPosition() == gfx::Point(0, 0)));
```

Comment:
This does not work for multi-monitors, where the position might not be `(0, 0)` when window is maximized in an external window.  I think you can just do `return GetBounds() == display.work_area()`, I have not tested it though.

Reply:
Thank you for catching that. I tested using `return GetBounds() == display.work_area()` on my two monitors and it seemed to function properly, so I replaced those lines in this next commit.